cpp/src/comms/ipc/ipc.cpp

```diff
@@ -1,9 +1,20 @@
+#include <arrow/result.h>
 #include <cudf/ipc.hpp>
+#include <cudf/utilities/error.hpp>
 
 CudaMessageReader::CudaMessageReader(arrow::cuda::CudaBufferReader* stream,
                                      arrow::io::BufferReader* schema)
   : stream_(stream), host_schema_reader_(schema){};
 
+arrow::Result<std::unique_ptr<arrow::ipc::Message>> CudaMessageReader::ReadNextMessage()
+{
+  std::unique_ptr<arrow::ipc::Message> message;
+  if (!ReadNextMessage(&message).ok()) {
+    return arrow::Status::Invalid("Failed to decode message");
+  }
+  return std::move(message);
+}
+
 arrow::Status CudaMessageReader::ReadNextMessage(std::unique_ptr<arrow::ipc::Message>* message)
 {
   if (host_schema_reader_ != nullptr) {
```

Comment:
The `arrow::ipc::ReadMessage` call just below still needs to be replaced right? It looks like [this](https://github.com/apache/arrow/blob/b058cf0d1c26ad7984c104bb84322cc7dcc66f00/cpp/src/arrow/ipc/message.h#L458-L460) one would work with `BufferReader`? Might be an opportunity to refactor the old `ReadNextMessage` here into the new one above since it doesn't like this is used anywhere else, unless something is using the Cython binding.

Reply:
Arrow's CudaMessageReader class still does it the old way, but I removed the old ReadMessage overload in our class :+1:

Reply:
Are you sure about `arrow::cuda::ReadMessage`? It looks like it was removed [here](https://github.com/apache/arrow/commit/ab93ba1f901ad6a2ce3d34404e6b46d2539507dbl)